features/300-request-verification/request-verification.md

```diff
@@ -2,91 +2,44 @@
 
 ## Motivation
 
-The gateway injects HTTP headers to tell the current authenticated user.
+Microservices may communicate with different components in Skygear:
+- Gateway in for incoming HTTP traffic.
+- Auth Gear for web-hook events.
 
-The auth gear sends webhooks.
+We need ways to verify the authenticity of communication with Gateway/Auth Gear.
 
-We need ways to verify the HTTP headers and the body are from the gateway
-and the auth gear.
+To verify authenticity of communication, we need to verify two property:
+- the identity of receiver (i.e. microservices)
+- the identity of sender (i.e. gateway/auth gear)
 
-## Special Headers
+The industry standard of securing and authenticating communication is to use
+mTLS. However, for simplicity, Skygear would use a simpler design to acheive
+this objective. This document describe Skygear's design of verification.
 
-`x-skygear-headers-signature`
+## Web-hook communication verification
```

Comment:
Web-hook could be outside of Skygear Cluster right?

If yes, readability-wise, this paragraph is a bit misleading.

Coz the last paragraph is talking about Communication between Gateway / Auth Gear, this paragraph is about web-hook which could be microservices or other HTTPS endpoint outside of cluster.

Should instead:
1. discuss the communication between gateway and auth gear first (move the last paragraph to here?)
2. make it clear, we have finished discussing gateway / authgear, now we discuss web-hook which is another thing?

Reply:
I intended that the heading levels should clarify the relationship between sections: there are only two second-level sections: Motivations, Web-hook communication verification, and Gateway communication verification.

I do think it is a bit confusing after a long section on web-hook, I'll move the gateway section before web-hook and clarify it further.

Reply:
> I intended that the heading levels should clarify the relationship between sections: there are only two second-level sections: Motivations, Web-hook communication verification, and Gateway communication verification.

Heading levels is not the problem here. Problem is usually people assume the first paragraph would be something like Introduction, so originally it didn't cover Webhook could be external services, it only mentions the microservices, gateway and auth gear, it could be quite misleading. 

The new force pushed one is better, thx.